File: Data Structures & Algorithms/max-water-container/submission-3.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        ans = 0


        l, r = 0, len(heights)-1


        while l<r:
            ans = max(ans, min(heights[r], heights[l])*(r-l))

            if heights[r]>heights[l]:
                l+=1
            elif heights[r]<heights[l]:
                r-=1

            else:
                l+=1
                r-=1

        return ans

        



        

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int l=0, ans=0; 
        unordered_set<char> check; 
        
        for(int r=0; r < s.size();r++ ){

            while(check.count(s[r])){
                check.erase(s[l]); 
                l++; 
            }
            check.insert(s[r]); 
            ans = max(ans, r-l+1); 
        }

        return ans; 
    }
};
